programs/bubblegum/program/src/lib.rs

```diff
@@ -1348,6 +1597,114 @@ pub mod bubblegum {
         )
     }
 
+    pub fn update_metadata<'info>(
+        ctx: Context<'_, '_, '_, 'info, UpdateMetadata<'info>>,
+        root: [u8; 32],
+        nonce: u64,
+        index: u32,
+        old_metadata: Option<MetadataArgs>,
+        new_name: Option<String>,
```

Comment:
Would it make sense to create a struct `UpdateMetadata` with all these parameters as `Option` fields?

Reply:
Yeah, we can do that if that's the preferred pattern

Reply:
Yes we prefer this pattern that we developed in TM which essentially looks like:
```
#[derive(Default)]
pub struct UpdateArgs {
   name: Option<String>,
   symbol: Option<String>,
   ...
}
```
Then we can do things like this to only change the name:
```
let update_args = UpdateArgs::default();
update_args.name = Some(String::from("new name"));
```

Reply:
Changed to use `UpdateArgs`